File: 003/solve.py
#!usr/bin/python

from math import sqrt

def isprime(n):
    for x in xrange(2, int(sqrt(n))+1):
        if n%x == 0:
            return False
    return True

def primes():
    y = 2
    while True:
        pr = y
        yield pr
        y += 1
        while isprime(y) is False:
            y += 1

number = 600851475143
P = primes()

while number != 1:
    x = P.next()
    if number%x == 0:
        number /= x
        result = x

print result

File: 003/solve.go
package main

import (
	"fmt"
	"math"
)

func isPrime(n int) bool {
	for i := 2; i < int(math.Sqrt(float64(n))+1); i++ {
		if n%i == 0 {
			return false
		}
	}

	return true
}

func main() {
	number := 600851475143
	max_prime := 0

	primes := make(chan int)

	go func() {
		for i := 2; ; i++ {
			if isPrime(i) {
				primes <- i
			}
		}
	}()

	for number != 1 {
		prime := <-primes
		if number%prime == 0 {
			number /= prime
			max_prime = prime
		}
	}

	fmt.Println(max_prime)
}
